game/src/core/systems/HudSystem.java

```diff
@@ -0,0 +1,78 @@
+package core.systems;
+
+import com.badlogic.gdx.scenes.scene2d.Group;
+
+import core.Game;
+import core.System;
+import core.components.UIComponent;
+
+public final class HudSystem extends System {
+
+    public HudSystem() {
+        super(UIComponent.class);
+    }
+
+    /* for dungeon Issue #727
+        @Override
+        private void addEntity(Entity entity) {
+            stage.addActor(entity.getComponent(UIComponent.class).map(UIComponent.class::cast).map(UIComponent::getDialog).get());
+        }
+    */
+
+    /* for dungeon Issue #727
+        @Override
+        public void removeEntity(Entity entity) {
+            entity.getComponent(UIComponent.class).map(UIComponent.class::cast).map(UIComponent::getDialog).get().remove();
+        }
+    */
+
+    @Override
+    public void execute() {
+        // Temp fix until either addEntity is available or removeEntity until dungeon Issue #727
+        // implemented
+        getEntityStream()
+                .forEach(
+                        x -> {
+                            Group d =
+                                    x.getComponent(UIComponent.class)
+                                            .map(UIComponent.class::cast)
+                                            .get()
+                                            .getDialog();
+                            Game.stage()
+                                    .ifPresent(
+                                            stage -> {
+                                                if (!stage.getActors().contains(d, true)) {
+                                                    stage.addActor(d);
+                                                }
+                                            });
+                        });
+
+        if (getEntityStream()
+                .anyMatch(
+                        x ->
+                                x.getComponent(UIComponent.class)
+                                        .map(UIComponent.class::cast)
+                                        .map(y -> y.isVisible() && y.isPauses())
+                                        .get())) {
+            pauseGame();
+        } else {
+            unpauseGame();
+        }
+    }
+
+    private void pauseGame() {
+        Game.systems.values().forEach(System::stop);
+    }
+
+    private void unpauseGame() {
+        Game.systems.values().forEach(System::run);
+    }
+
+    /** HudSystem can´t be paused */
+    @Override
+    public void stop() {}
+
+    /** HudSystem is always running no need to log it */
```

Comment:
den kommentar verstehe ich im zusammenhang mit der methode nicht wirklich? was hat logging mit `run` zu tun?

Reply:
run in `core.System` `LOGGER.info(this.getClass().getName() + " is running");` welches bei einem System, welches nie pausiert wird, unnötig meiner Meinung nach dem Log zu spammt. Vielleicht bin ich da auch etwas komisch und es ist gewollt das es dann auch immer im Log steht. Oder vielleicht was für #683 wo nur bei Status Änderung ein Logeintrag geschrieben wird.

Reply:
Das hat mehrere Kompomenten:

(a) Vielleicht macht das Logging/Spamming Sinn, aber man sollte es nicht unter "Info" packen sondern unter "Finer" oder so? Ist aber vermutlich wirklich eine Sache für #683 ... (Über den Sinn kann ich aber nur spekulieren.)
(b) Der Javadoc-Kommentar erklärt das halt nicht - hier wäre Deine Erklärung von oben eine gute Ergänzung für den Javadoc.


Die Frage ist: Wie gehen wir damit um? Vielleicht sollte man das Stückchen Code hier wieder löschen und damit das "Spamming" beibehalten und in #683 diese Diskussion mit einkippen? Wenn dort nämlich beschlossen wird, den Status nie zu loggen oder immer zu loggen auf einem anderen Level, dann ist dieses System hier "tot" und wird vielleicht nicht beachtet.

Reply:
Stimmt klingt sinnvoll. Werde in #683 erwähnen, was wir hier grob bedacht haben und dass dieser Gedanke nicht verloren geht.